service/worker/service.go

```diff
@@ -447,19 +450,32 @@ func (svc *Service) handleNodeBindingUpdates(ctx context.Context, newBindingPair
 	}
 
 	for _, old := range oldBindings {
+		oldKey := old.Key()
 		found := false
 		for _, newBinding := range newBindings {
-			if old.Key() == newBinding.Key() {
+			if oldKey == newBinding.Key() {
 				found = true
 				break
 			}
 		}
 
 		if !found {
-			logger.Ctx(ctx).Infow("binding removed", "key", old.Key())
-			handler := svc.pushHandlers[old.Key()]
-			handler.Stop()
-			delete(svc.pushHandlers, old.Key())
+			func() {
+				svc.pushHandlersLock.Lock()
```

Comment:
instead of mutex can the datastructure itself be like that which supports synchronised read/writes

Reply:
yup. Involves a bit of refactoring. Will take ut up in subsequent PRs. Tracking [here](https://github.com/razorpay/metro/issues/126)